records_mover/records/validated_records_hints.py

```diff
@@ -0,0 +1,151 @@
+            # in a list of specific optional strings.  It's wrong;
+            # that's perfectly safe
+            try:
+                i = valid_list.index(x)  # type: ignore
+                return valid_list[i]
+            except ValueError:
+                cant_handle_hint(fail_if_cant_handle_hint=fail_if_cant_handle_hint,
+                                 hint_name=hint_name,
+                                 hints=hints)
+                return default
+
+        def validate_compression() -> HintCompression:
+            return validate_literal(VALID_COMPRESSIONS,
+                                    'compression',
+                                    default=None)
+
+        def validate_quoting() -> HintQuoting:
+            return validate_literal(VALID_QUOTING,
+                                    'quoting',
+                                    default='minimal')
+
+        def validate_escape() -> HintEscape:
+            return validate_literal(VALID_ESCAPE,
+                                    'escape',
+                                    default='\\')
+
+        def validate_encoding() -> HintEncoding:
+            return validate_literal(VALID_ENCODING,
+                                    'encoding',
+                                    default='UTF8')
+
+        def validate_dateformat() -> HintDateFormat:
+            return validate_literal(VALID_DATEFORMATS,
+                                    'dateformat',
+                                    default='YYYY-MM-DD')
+
+        def validate_timeonlyformat() -> HintTimeOnlyFormat:
+            return validate_literal(VALID_TIMEONLYFORMATS,
+                                    'timeonlyformat',
+                                    default="HH24:MI:SS")
+
+        def validate_datetimeformattz() -> HintDateTimeFormatTz:
+            return validate_literal(VALID_DATETIMEFORMATTZS,
+                                    'datetimeformattz',
+                                    default="YYYY-MM-DD HH24:MI:SSOF")
+
+        def validate_datetimeformat() -> HintDateTimeFormat:
+            return validate_literal(VALID_DATETIMEFORMATS,
+                                    'datetimeformat',
+                                    default="YYYY-MM-DD HH24:MI:SS")
+
+        header_row = validate_boolean('header-row')
+        field_delimiter = validate_string('field-delimiter')
+        compression = validate_compression()
+        record_terminator = validate_string('record-terminator')
+        quoting = validate_quoting()
+        quotechar = validate_string('quotechar')
+        doublequote = validate_boolean('doublequote')
+        escape = validate_escape()
+        encoding = validate_encoding()
+        dateformat = validate_dateformat()
+        timeonlyformat = validate_timeonlyformat()
+        datetimeformattz = validate_datetimeformattz()
+        datetimeformat = validate_datetimeformat()
+        # TODO: After one create functions
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker